app/src/main/java/com/android/streetworkapp/ui/park/AddEvent.kt

```diff
@@ -64,11 +67,32 @@ import java.util.concurrent.TimeUnit
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun AddEventScreen(
-    navigationActions: NavigationActions
-) { // TODO: add future ParkViewModel and EventViewModel
+    navigationActions: NavigationActions,
+    parkViewModel: ParkViewModel,
+    eventViewModel: EventViewModel,
+    userViewModel: UserViewModel
+) {
 
-  // used until we have the corresponding viewModel TODO: update with new viewModel
-  val event = Event("unknown", "unknown", "unknown", 0, 2, Timestamp(0, 0), "unknown")
+  val eid = eventViewModel.getNewEid()
+  val event =
+      Event(
+          eid,
+          "unknown",
+          "unknown",
+          0,
+          EventConstants.MIN_NUMBER_PARTICIPANTS,
+          Timestamp(0, 0),
+          "unknown")
+
+  val owner = userViewModel.currentUser.value?.uid
```

Comment:
I see your concern. However, I think it is better to use the uid, since we can be certain that it is unique in the database. To use the username safely, we will first need a mechanism that ensure that each username is unique, and we do not have it yet. 
This is why I still think that we should use the uid. Do you agree?

Reply:
I see where you are coming from, a compromise might be to store the uid as well as the username, I say that mainly because when fetch back the events to display them, it wouldn't make sense to display the uid instead of the username on an event overview screen, but this can be changed later when all the user logic has been implemented